app/scripts/components/changeToDoItem/changeToDoItem.template.html

```diff
@@ -0,0 +1,5 @@
+<span ng-switch="$ctrl.editMode">
+    <input ng-switch-when="true" type="{{$ctrl.fieldType}}" ng-model="$ctrl.fieldValue">
+</span>
+<button ng-click="$ctrl.handleModeChange()">{{$ctrl.editMode ? 'Save' : 'Edit'}}</button>
```

Comment:
Could you please move the expression from double braces to a separate method in the controller?